### PR TITLE
SAAS-3259: Added fraudDetectorDataEncrypted plugin and spec

### PR DESCRIPTION
**Pull request by m-nouman-suleman**
## Task
- https://scalock.atlassian.net/browse/SAAS-3259

### CODE REVIEW

File: plugins/aws/frauddetector/fraudDetectorDataEncrypted.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                return rcb();
+            }
+
+            if (!listDetectors.data.length) {
+                helpers.addResult(results, 0, 'No Fraud Detectors found', region);
+                return rcb();
+            }
+
+            var fraudDetectorsEncryptionKey = helpers.addSource(cache, source,
+                ['frauddetector', 'getKMSEncryptionKey', region, 'data']);
```
Comment (AkhtarAmir) [inline plugins/aws/frauddetector/fraudDetectorDataEncrypted.js:52]:
```suggestion
                ['frauddetector', 'getKMSEncryptionKey', region]);
```

File: plugins/aws/frauddetector/fraudDetectorDataEncrypted.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
```
Comment (AkhtarAmir) [inline plugins/aws/frauddetector/fraudDetectorDataEncrypted.js:53]: Add unknown check

File: plugins/aws/frauddetector/fraudDetectorDataEncrypted.spec.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            fraudDetectorDataEncrypted.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should UNKNOWN if unable to list Fraud Detector data', function (done) {
+            const cache = createCache(null, null, null);
```
Comment (AkhtarAmir) [inline plugins/aws/frauddetector/fraudDetectorDataEncrypted.spec.js:146]:
```suggestion
            const cache = createCache();
```

**Review by AkhtarAmir — APPROVED** [review]

File: plugins/aws/frauddetector/fraudDetectorDataEncrypted.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                let encryptionKey = fraudDetectorsEncryptionKey.data.kmsEncryptionKeyArn;
+                var keyId = encryptionKey.split('/')[1] ? encryptionKey.split('/')[1] : encryptionKey;
+
+                var describeKey = helpers.addSource(cache, source,
+                    ['kms', 'describeKey', region, keyId]);
+
+                if (!describeKey || describeKey.err || !describeKey.data || !describeKey.data.KeyMetadata) {
+                    helpers.addResult(results, 3,
+                        `Unable to query KMS key: ${helpers.addError(describeKey)}`,
+                        region, encryptionKey);
```
Comment (dipsubha06) [inline plugins/aws/frauddetector/fraudDetectorDataEncrypted.js:80]:
```suggestion
                        region, encryptionKey);
                    return rcb();    
```

**Pull request merged**